### ISSUE TITLE
Alias of list<T> tolerates null items

### ISSUE THREAD

**Issue by iamdanfox** [palantir/conjure-java#104]
## What happened

The following Conjure type:

```
MyList
  alias: list<string>
```

Deserialized from `[null, null]` without any errors, leading to nasty nulls in my codebase:
```java
System.out.println(ObjectMappers.newServerObjectMapper()
    .readValue("[null, null]", MyList.class)); // prints [null, null]
```

## What should have happened

These nulls should have been rejected.  This matches guava's ImmutableList behaviour, which nullchecks every item you add (even if you do `copyOf`).

**Comment by ferozco** [comment]
Closed by #676

**Comment by carterkozak** [comment]
there's still a case here where MyList.of() can accept a list with nulls, but that's less dangerous because remote servers will not accept it.
